climpred/metrics.py

```diff
@@ -41,16 +43,95 @@ def _get_norm_factor(comparison):
         fac = 2
     else:
         raise KeyError('specify comparison to get normalization factor.')
     return fac
 
 
+def _display_metric_metadata(self):
+    summary = '----- Metric metadata -----\n'
+    summary += f'Name: {self.name}\n'
+    summary += f'Longname: {self.longname}\n'
+    summary += f'Alias: {self.aliases}\n'
+    # positively oriented
+    if self.is_positive:
+        summary += 'Orientation: positive\n'
+    else:
+        summary += 'Orientation: negative\n'
+    # probabilistic or deterministic
+    if self.is_probabilistic:
+        summary += 'Kind: probabilistic\n'
+    else:
+        summary += 'Kind: deterministic\n'
+    summary += f'Power of units: {self.unit_power}\n'
+    summary += f'Minimum skill {self.min}\n'
+    summary += f'Maximum skill {self.max}\n'
+    summary += f'Perfect skill {self.perfect}\n'
+    summary += f'Strictly proper score: {self.proper}\n'
+    # doc
+    summary += f'Function: {self.function.__doc__}\n'
+    return summary
+
+
+class Metric:
+    'Master class for all metrics.'
+
+    def __init__(
+        self,
+        name,
+        function,
+        is_positive,
+        is_probabilistic,
+        longname=None,
+        aliases=None,
+        unit_power=1,
+        min=None,
+        max=None,
+        perfect=None,
+        proper=None,
+    ):
+        """Metric initialization.
+
+        Args:
+            name (str): name of metric.
+            function (function): metric function.
+            is_positive (bool): Is metric positively oriented? Higher metric
+             values means higher skill.
+            is_probabilistic (bool): Is metric probabilistic? `False` means
+             deterministic.
+            unit_power (float, int): Power of the input unit for unit of skill.
+            longname (str, optional): Longname of metric. Defaults to None.
+            aliases (list, optional): Allowed aliases for this metric. Defaults to None.
+            min (float, optional): Minimum skill for metric. Defaults to None.
+            max (float, optional): Maxmimum skill for metric. Defaults to None.
+            perfect (float, optional): Perfect skill for metric. Defaults to None.
+            proper (bool, optional): Is strictly proper skill score? Defaults to None.
+
+        Returns:
+            Metric: metric class Metric.
+
+        """
+        self.name = name
+        self.longname = longname
+        self.function = function
+        self.is_positive = is_positive
+        self.is_probabilistic = is_probabilistic
+        self.aliases = aliases
+        self.unit_power = unit_power
+        self.min = min
+        self.max = max
+        self.perfect = perfect
+        self.proper = proper
+
+    def __repr__(self):
+        """Show metadata of metric class."""
+        return _display_metric_metadata(self)
+
+
 def _pearson_r(forecast, reference, dim=None, **metric_kwargs):
     """
     Calculate Person's Anomaly Correlation Coefficient (ACC).
```

Comment:
I would drop "Calculate" and bump this up to just right of the """. Also `Person's` is a typo here.